src/main.rs

```diff
@@ -0,0 +1,253 @@
+    read_wire_and_operate!(input_file, msg, {
+        serde_json::to_writer(output, &msg)
+            .expect("failed to serialize to JSON")
+    });
+}
+
+/// Deserializes a header + message in wire format from `input_file` and then prints the message
+/// to the console.
+fn print_message(input_file: &str) {
+    read_wire_and_operate!(input_file, msg, { eprintln!("{:?}", msg) });
+}
+
+fn main() {
+    let app = App::new("Manticore Tool")
+        .version("0.1")
+        .author("lowRISC contributors")
+        .about("Command line tool for Manticore library")
+        .setting(AppSettings::ArgRequiredElseHelp)
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .arg(
+            Arg::with_name("v")
+                .short("v")
+                .multiple(true)
+                .help("Sets the level of verbosity"),
+        )
+        .subcommand(
+            SubCommand::with_name("fromjson")
+                .about("Generate a message from JSON and serialize it")
+                .arg(
+                    Arg::with_name("cmdtype")
+                        .short("t")
+                        .long("cmdtype")
+                        .help("Command type")
+                        .required(true)
+                        .takes_value(true),
+                )
+                .arg(
+                    Arg::with_name("isrequest")
+                        .short("r")
+                        .long("isrequest")
+                        .help("Whether the message is a request"),
+                )
+                .arg(
+                    Arg::with_name("input")
+                        .short("i")
+                        .long("input")
+                        .help("JSON input file with message data")
+                        .required(true)
+                        .takes_value(true),
+                )
+                .arg(
+                    Arg::with_name("output")
+                        .short("o")
+                        .long("output")
+                        .help("output file for serialized message")
+                        .required(true)
+                        .takes_value(true),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("tojson")
+                .about("Deserialize a message and store it as JSON")
+                .arg(
+                    Arg::with_name("input")
+                        .short("i")
+                        .long("input")
+                        .help("input file containing serialized message")
+                        .required(true)
+                        .takes_value(true),
+                )
+                .arg(
+                    Arg::with_name("output")
+                        .short("o")
+                        .long("output")
+                        .help("output file for message data as JSON")
+                        .required(true)
+                        .takes_value(true),
+                ),
```

Comment:
Do you think that, instead of using file arguments, we could use stdin/stdout? That would make the commandline parsing simpler, and it would mean you could call it like
```
manticore-tool deserialize < blah.bin
```
instead of having `print` and `tojson` be separate code. The same with `fromjson`, since you could then write
```
manticore-tool serialize > blah.bin <<< '{json here}'
```

Reply:
I had thought of that, but I don't like the idea of sending raw binary data onto the console by default. You can already do `-i /dev/stdin` and `-o /dev/stdout`. We could look into adding `-i -` and `-o -` as aliases, but I don't think that's necessary. I'd leave it as is if that's ok.

Reply:
Let's leave it like this for now, but we should re-visit this interface to make it a bit less redundant.